src/GslCore/AstTypes.fs

```diff
@@ -352,6 +367,45 @@ and Roughage = {locus: Node<L2Id> option; marker: Node<string> option; parts: No
             | Some(mw) -> Some(mw.x)
         | Some(mw) -> Some(mw.x) // Yes there is a marker attached to the locus knockout
 
+/// Helper function for pushing new source code positions onto stack during function expansion
+/// Most recently (first) postition refers to highest level function invocation.
+let pushPos (newPos:SourcePosition) (x:AstNode) =

```

Comment:
I think this should be refactored to to prepend an entire new list of positions; that will simplify the sole use of this code in the function expansion section.  Basically pushPosNode becomes, say, prependPositionsNode:
```
let prependPositionsNode (newPositions: SourcePosition list) (node: Node<'T>) =
    {node with positions=newPositions@node.positions}
```
Then this function gains the signature `SourcePosition list -> AstNode -> AstNode` and the `List.rev` and `List.fold` in the function renumbering function should disappear.  I *think* this gives us the correct order including nested positions.